### PR TITLE
Add humidity tag R2.1 and support select bridge

### PR DESCRIPTION
**Pull request by blavka**

### CODE REVIEW

File: src/application.c [hunk 1]
```diff
@@ -175,6 +178,21 @@ static void _application_wait_start_string(void)
     }
 }
 
+static void _application_bridge_led_flashes(bc_bridge_t *bridge, int count)
+{
+    int i;
+
+    bc_bridge_led_set_state(bridge, BC_BRIDGE_LED_STATE_OFF);
+
+    for (i=0; i<count; i++)
```
Comment (hubpav) [inline src/application.c]: Missing spaces, follow coding style standard

File: src/application.c [hunk 2]
```diff
@@ -175,6 +178,21 @@ static void _application_wait_start_string(void)
     }
 }
 
+static void _application_bridge_led_flashes(bc_bridge_t *bridge, int count)
```
Comment (hubpav) [inline src/application.c]: Better name is _application_bridge_flash_led

File: src/application.c [hunk 3]
```diff
@@ -55,6 +56,26 @@ void application_init(application_parameters_t *parameters)
...
+        memset(devices, 0, sizeof(devices));
+        uint8_t device_count = sizeof(devices) / sizeof(bc_bridge_device_info_t);
+
+        if (!bc_bridge_scan(devices, &device_count))
+        {
+            bc_log_error("application_loop: call failed: bc_bridge_scan");
+            exit(EXIT_FAILURE);
+        }
+        int i;
+        printf("id path \n");
```
Comment (hubpav) [inline src/application.c]: Why is there dummy space before \n? Please change format to printf("ID" "\t" "Path" "\n");

File: src/application.c [hunk 4]
```diff
@@ -55,6 +56,26 @@ void application_init(application_parameters_t *parameters)
...
+        for (i = 0; i < device_count; i++)
```
Comment (hubpav) [inline src/application.c]: Please write for (int i = 0; ......)

File: src/application.c [hunk 5]
```diff
@@ -55,6 +56,26 @@ void application_init(application_parameters_t *parameters)
...
+        bc_bridge_device_info_t devices[16];
+        memset(devices, 0, sizeof(devices));
+        uint8_t device_count = sizeof(devices) / sizeof(bc_bridge_device_info_t);
+
+        if (!bc_bridge_scan(devices, &device_count))
+        {
+            bc_log_error("application_loop: call failed: bc_bridge_scan");
+            exit(EXIT_FAILURE);
+        }
+        int i;
```
Comment (hubpav) [inline src/application.c]: You can delete this line

File: src/application.c [hunk 6]
```diff
@@ -55,6 +56,26 @@ void application_init(application_parameters_t *parameters)
...
+        printf("id path \n");
+        for (i = 0; i < device_count; i++)
+        {
+            printf("% 2d %s \n", i, devices[i].usb_path);
```
Comment (hubpav) [inline src/application.c]: Please change to printf("%d" "\t" "%s" "\n", i, devices[i].usb_path);

File: src/application.c [hunk 7]
```diff
@@ -63,21 +84,24 @@ void application_init(application_parameters_t *parameters)
     memset(&bridge, 0, sizeof(bridge));
 }
 
-void application_loop(bool *quit)
+void application_loop(bool *quit, application_parameters_t *parameters)
 {
     *quit = false;
 
-    bc_bridge_device_info_t devices[10];
+    bc_bridge_device_info_t devices[16];
```
Comment (hubpav) [inline src/application.c]: Please check that enumaration function will never overflow outside this area.

File: src/application.c [hunk 8]
```diff
@@ -88,14 +112,40 @@ void application_loop(bool *quit)
...
+            if (parameters->order > -1)
+            {
+                if (parameters->order < device_count)
+                {
+                    selected_devices = parameters->order;
+                }
+            }
+            else if (parameters->path != NULL)
+            {
+                for (i = 0; i < device_count; i++)
```
Comment (hubpav) [inline src/application.c]: for (int i.....)

File: src/application.c [hunk 9]
```diff
@@ -88,14 +112,40 @@ void application_loop(bool *quit)
...
+                {
+                    if (strcmp(devices[i].usb_path, parameters->path) != -1)
```
Comment (hubpav) [inline src/application.c]: This is a bug actually - you should check for == 0 for strcmp...

File: src/application.c [hunk 10]
```diff
@@ -88,14 +112,40 @@ void application_loop(bool *quit)
...
+        {
             break;
         }
 
         bc_os_task_sleep(5000);
 
     }
 
-    if (!bc_bridge_open(&bridge, &devices[0]))
+    if (!bc_bridge_open(&bridge, &devices[selected_devices]))
```
Comment (hubpav) [inline src/application.c]: It should not be plural - it is selected_device

File: src/application.c [hunk 11]
```diff
@@ -184,7 +233,7 @@ static void _application_bridge_led_flashes(bc_bridge_t *bridge, int count)
 
     bc_bridge_led_set_state(bridge, BC_BRIDGE_LED_STATE_OFF);
 
-    for (i=0; i<count; i++)
+    for (i = 0; i < count; i++)
```
Comment (hubpav) [inline src/application.c]: Correct :+1: :-)

File: src/main.c [hunk 12]
```diff
@@ -100,6 +109,26 @@ static error_t parse_opt(int key, char *arg, struct argp_state *state)
...
         }
+        case OPTION_LIST:
+        {
+            application_parameters->show_list = true;
+            break;
+        }
+        case OPTION_ID:
+        {
+            char *end;
+            application_parameters->order = (int) strtol(arg, &end, 10);
```
Comment (hubpav) [inline src/main.c]: The name is not consistent, please change to application_parameters->device_id

File: CHANGELOG.adoc [hunk 13]
```diff
@@ -2,6 +2,12 @@
 :gh-url: https://github.com/bigclownlabs/bc-bridge
 
 == link:{gh-url}/tree/v1.1.0[1.1.0] (unreleased)
+Added::
+* Support for Humidity TAG R2.1
+* New command options:
+** `--dev-list` Show list of available devices
```
Comment (hubpav) [inline CHANGELOG.adoc]: I strongly suggest to keep command line options simple - thus:
--list
--id
--path

File: src/application.c (hunk 2 above)
Comment (blavka) [inline src/application.c]: Ok

File: src/application.c (hunk 4 above)
Comment (blavka) [inline src/application.c]: ok

File: src/application.c (hunk 3 above)
Comment (blavka) [inline src/application.c]: ok

File: src/application.c (hunk 5 above)
Comment (blavka) [inline src/application.c]: ok

File: src/application.c (hunk 7 above)
Comment (blavka) [inline src/application.c]: Function bc_bridge_scan get two parameter `devices`,` devices_count`,  `devices_count` in input contains max size devices array, but as output contains how how many were found

File: src/application.c (hunk 8 above)
Comment (blavka) [inline src/application.c]: ok

File: src/application.c (hunk 10 above)
Comment (blavka) [inline src/application.c]: repaired

File: src/main.c (hunk 12 above)
Comment (blavka) [inline src/main.c]: changed

File: CHANGELOG.adoc (hunk 13 above)
Comment (blavka) [inline CHANGELOG.adoc]: Ok change on --list and --dev

File: src/main.c [hunk 14]
```diff
@@ -40,9 +38,9 @@ int main(int argc, char **argv)
         {
             .furious_mode = false,
             .log_level = BC_LOG_LEVEL_WARNING,
-            .show_list = false,
-            .order = -1,
-            .path = NULL
+            .dev_list = false,
+            .dev_num = -1,
```
Comment (hubpav) [inline src/main.c]: The variable should be dev_id now

File: src/application.c (hunk 7 above)
Comment (hubpav) [inline src/application.c]: Okay thanks

File: src/main.c (hunk 14 above)
Comment (blavka) [inline src/main.c]: Fix

**Pull request merged**